packages/flutter/test/material/dropdown_form_field_test.dart

```diff
@@ -715,4 +715,52 @@ void main() {
     expect(value, equals('two'));
     expect(dropdownButtonTapCounter, 2); // Should not change.
   });
+
+  testWidgets('DropdownButtonFormField should re-render if value param changes', (WidgetTester tester) async {
+    String currentValue = 'two';
+    final GlobalKey<FormFieldState<String>> key = GlobalKey<FormFieldState<String>>();
+
+    await tester.pumpWidget(
+      StatefulBuilder(
+        builder: (BuildContext context, StateSetter setState) {
+          return MaterialApp(
+            home: Material(
+              child: DropdownButtonFormField<String>(
+                key: key,
+                value: currentValue,
+                hint: const Text('Select Value'),
+                decoration: const InputDecoration(
+                    prefixIcon: Icon(Icons.fastfood)
+                ),
```

Comment:
These params (`hint` and `decoration`) do not seem necessary. It would be better to remove them to make the test clearer.